scripts/update-changelog-header-for-release.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
```

Comment:
what happens if the Unreleased section was already modified before, e.g. this already ran and we're running release workflow again?

will it fail or will do nothing silently?

Reply:
The script will output a message like `CHANGELOG already has a header defined for 1.16.0, no changes made to changelog.` (example: https://github.com/AgustinBettati/terraform-provider-mongodbatlas/actions/runs/8638524825/job/23683156506).